### PR TITLE
feat!: Adopt gRPC

### LINKED ISSUE #38: Support user-defined output
cerbos/cerbos#1594

### PR DESCRIPTION
**Pull request by oguzhand95**
PHP SDK utilized HTTP API due to some limitations related to code generation. Now, using `buf` and some newly discovered tricks, it is possible to generate php code from the gRPC protobuf definitions.

This PR re-designs the SDK to use `gRPC` with some changes breaking backward compatibility.

### Supported [PHP versions](https://www.php.net/supported-versions.php)
- `8.1`
- `8.2`

resolves #38

### CODE REVIEW

**Review by charithe — APPROVED** [review]
LGTM. Since this is a major breaking change, before merging this to `main`, create a new branch from `main` so that we can continue to support existing users with security patches etx. until they migrate to the new version.

We should also add a migration guide explaining how to convert existing code to work with the new SDK as well.

File: Makefile [hunk 1]
```diff
@@ -18,4 +26,12 @@ validate-composer:
 test:
 	docker-compose up -d
 	vendor/bin/phpunit --no-coverage
-	docker-compose down
\ No newline at end of file
+	docker-compose down
+
+.PHONY: generate-proto-code
+generate-proto-code:
```
Comment (charithe) [inline Makefile:32]: Are you just changing version 2 to 3? It probably could be done with `sed`.

File: README.md [hunk 2]
```diff
@@ -58,65 +59,173 @@ if ($resultEntry->isAllowed("approve")) { // returns true if `approve` action is
...
+This part of the `README.md` describes the changes made between PHP SDK `v0.1.x` version and this version.
+
+Since `v0.1.x` version(s) used HTTP API and this version use the gRPC API, there are a couple of steps to take.
+
+## gRPC
+
+`gRPC` extension required to be installed on the development and production environment.
+
+There is a [nice guide](https://cloud.google.com/php/grpc) from Google describing how to download and enable the `gRPC`
+extension.
```
Comment (charithe) [inline README.md:145]: Hmm... wouldn't this require an administrator to update the server installation ? It looks like it's no longer a matter of just deploying some code. Is that normal in PHP deployments?

Comment (oguzhand95) [reply README.md:145]: Yes, I think so.
You are even able to state what extensions are required in composer.json. ([link](https://github.com/cerbos/cerbos-sdk-php/pull/41/files#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R38-R39))

**Pull request merged**